Doc/tools/check-warnings.py

```diff
@@ -24,28 +26,177 @@
     "venv",
 }
 
-PATTERN = re.compile(r"(?P<file>[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)")
+# Regex pattern to match the parts of a Sphinx warning
+WARNING_PATTERN = re.compile(
+    r"(?P<file>([A-Za-z]:[\\/])?[^:]+):(?P<line>\d+): WARNING: (?P<msg>.+)"
+)
 
+# Regex pattern to match the line numbers in a Git unified diff
+DIFF_PATTERN = re.compile(
+    r"^@@ -(?P<linea>\d+)(?:,(?P<removed>\d+))? \+(?P<lineb>\d+)(?:,(?P<added>\d+))? @@",
```

Comment:
It seems we don't use ``linea`` and ``removed`` at all, could we simplify the pattern?

Reply:
Yeah, they aren't used right now, but we might in the future, and they don't really do any harm and make the regex more literate and self-documenting.